service/controller/resource/scrapeconfigs/resource.go

```diff
@@ -98,12 +100,16 @@ func toSecret(v interface{}, config Config) (*corev1.Secret, error) {
 	var workloadClusterETCDDomain string = ""
 	if "workload_cluster" == key.ClusterType(v) {
 		clusterID := key.ClusterID(cluster)
+		// Try to get the etcd url from the Giant Swarm way
 		service, err := config.K8sClient.K8sClient().CoreV1().Services(clusterID).Get(context.Background(), "master", metav1.GetOptions{})
-		if err != nil {
+		if apierrors.IsNotFound(err) {
+			// fallthrough, we need to get it from capi but ignore it for now
```

Comment:
So, if we ignore it here, where is it being populated?

Reply:
Empty, as we do not have the etcd url for now.

Reply:
We will need to fix it at a later time but by default the etcd's metric endpoint is not used

Reply:
If this need to be fixed later maybe label this comment with a `TODO`.